internal/cli/cmd/dac/preview/preview.go

```diff
@@ -146,13 +150,32 @@ func (o *option) computeEphemeralDashboardName(dashboardName string) string {
 }
 
 func (o *option) buildPreviewResponse(dashboard *modelV1.Dashboard, tmpDashboard *modelV1.EphemeralDashboard) previewResponse {
+	project := tmpDashboard.Metadata.Project
+	name := dashboard.Metadata.Name
+
 	finalURL := &url.URL{}
 	*finalURL = *o.apiClient.RESTClient().BaseURL
-	finalURL.Path = fmt.Sprintf("/%s/%s/%s/%s", utils.PathProject, tmpDashboard.Metadata.Project, utils.PathEphemeralDashboard, tmpDashboard.Metadata.Name)
+	finalURL.Path = fmt.Sprintf("/%s/%s/%s/%s", utils.PathProject, project, utils.PathEphemeralDashboard, tmpDashboard.Metadata.Name)
+
+	// Check if the dashboard already exists to provide the "current" URL
+	currentURL := &url.URL{}
+	_, err := o.apiClient.V1().Dashboard(project).Get(name)
```

Comment:
@Nexucis it looks like I shouldn't query the API this way if I want the unit tests to pass.. any idea how to fix that?
Btw we are using this syntax for other commands (`dac diff` & `project`..) but they either have no unit tests, or tests that don't cover such instruction ^^'

Reply:
actually you can, but you need to inject the fake client available in the unit test with the attribute `APIClient`:

```go
APIClient:       fakeapi.New(),
```